__tests__/babel.test.js

```diff
@@ -0,0 +1,275 @@
+        path.join(dirPath, '_underreact-site', 'underreact-assets')
+      );
+      const mainFileContent = getChunkContent(tree, 'main');
+      expect(mainFileContent).toMatch(/return"UNDERREACT_TEST_RETURN_VALUE"/);
+    });
+});
+
+test('Browserslist takes development env value', () => {
+  const fixture = generateFixture({
+    src: {
+      'entry.js': `
+        window.foo = ()=>'UNDERREACT_TEST_RETURN_VALUE';
+      `
+    },
+    'underreact.config.js': `
+        module.exports = { 
+            browserslist: {
+                production: ['ie 11'],
+                development: ['chrome 67']
+            }
+        }
+    `
+  });
+  return fixture
+    .then(dirPath => {
+      return tap(commandBuild({ cwd: dirPath, args: ['-m=development'] })).then(
+        () => dirPath
+      );
+    })
+    .then(dirPath => {
+      const tree = dirTree(
+        path.join(dirPath, '_underreact-site', 'underreact-assets')
+      );
+      const mainFileContent = getChunkContent(tree, 'main');
+      expect(mainFileContent).toMatch(/=> 'UNDERREACT_TEST_RETURN_VALUE'/);
+    });
+});
+
+test('Reads browserslist from package.json', () => {
+  const fixture = generateFixture({
+    src: {
+      'entry.js': `
+            window.foo = () => 'UNDERREACT_TEST_RETURN_VALUE';
+          `
+    },
+    'package.json': `{
+            "browserslist": [
+                "chrome 67"
+            ]
+        }`
+  });
+  return fixture
+    .then(dirPath => {
+      return tap(commandBuild({ cwd: dirPath })).then(() => dirPath);
+    })
+    .then(dirPath => {
+      const tree = dirTree(
+        path.join(dirPath, '_underreact-site', 'underreact-assets')
+      );
+      const mainFileContent = getChunkContent(tree, 'main');
+      expect(mainFileContent).toMatch(/=>"UNDERREACT_TEST_RETURN_VALUE"/);
+    });
+});
+
+test('Throws an error when using flow without flow plugin', () => {
+  const build = generateFixture({
+    src: {
+      'entry.js': `
+        // @flow
+        var t: number = 5;
+      `
+    },
+    'babel.config.js': `
+        module.exports = {
+            presets: []
+        }
+      `
+  }).then(dirPath => commandBuild({ cwd: dirPath }));
+
+  expect(build).rejects.toMatch(/SyntaxError:/);
+  return expect(build).rejects.toMatch(/ERROR: Compilation error./);
+});
+
+test('Parses flow when using correct plugin', () => {
+  const babelPresetMapbox = require.resolve('../packages/babel-preset-mapbox');
+  const babelPluginFlowStrip = require.resolve(
+    '@babel/plugin-transform-flow-strip-types'
+  );
+  const build = generateFixture({
+    src: {
+      'entry.js': `
+        // @flow
+        var t: number = 5;
+      `
+    },
+    'babel.config.js': `
+        module.exports = {
+            presets: ['${babelPresetMapbox}'],
+            plugins: ['${babelPluginFlowStrip}']
+        }
+      `
+  }).then(dirPath => commandBuild({ cwd: dirPath }));
+
+  expect(build).resolves.toMatch(/Using an external Babel config/);
```

Comment:
I think you need to `return` uses of `resolves` and `rejects` in order to properly handle errors.